File: src/main/java/fr/marstech/mtlinkspray/controller/ViewRootController.java
package fr.marstech.mtlinkspray.controller;

import fr.marstech.mtlinkspray.enums.ViewNameEnum;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.servlet.ModelAndView;

import static fr.marstech.mtlinkspray.enums.ViewNameEnum.HOME;

@Slf4j
@Controller
public class ViewRootController implements ThymeleafViewControllerInterface {

    private static final ViewNameEnum viewNameEnum = HOME;

    @Override
    public ModelAndView getModelAndView() {
        log.info("getModelAndView:{}", viewNameEnum);
        return getModelAndView(viewNameEnum);
    }

    @GetMapping("/")
    public ModelAndView getHome() {
        log.info("getHome");
        return getModelAndView();
    }
}

File: src/main/kotlin/fr/marstech/mtlinkspray/controller/view/ViewRootController.kt
package fr.marstech.mtlinkspray.controller.view

import fr.marstech.mtlinkspray.enums.ViewNameEnum.HOME
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.servlet.ModelAndView

@Controller
@RequestMapping("/")
class ViewRootController : ThymeleafViewControllerInterface {

    override fun getModelAndView(): ModelAndView = getModelAndView(HOME)

    @GetMapping("index")
    fun getIndex(): String = getRedirectUrl("/")

    @GetMapping
    fun getHome(): ModelAndView = getModelAndView()
}
